### PR TITLE
Offboard temp cal: check topic instance

### PR DESCRIPTION
**Pull request by bresch**
relates to https://github.com/PX4/Firmware/pull/14776

I found out that the script was assuming that the multi topics instance are appearing in order but it isn't always the case. In my case, the instance 1 appears before the 0.
Before fix (the 2nd IMU was assumed 0)
DeepinScreenshot_select-area_20200513132302
After fix (primary is 0)
DeepinScreenshot_select-area_20200513132309

### CODE REVIEW

**Comment by dagar** [comment]
>  In my case, the instance 1 appears before the 0.
> Before fix (the 2nd IMU was assumed 0)

I'm not sure I follow, what do you mean "instance 1 appears before the 0"? Are you saying the instance numbering of the logged topic (ulg) is inconsistent with the system?

File: Tools/process_sensor_caldata.py [hunk 1]
```diff
@@ -48,57 +48,51 @@ def is_valid_directory(parser, arg):
...
             print('found baro 0 data')
-            num_baros = 1
-        if sensor_instance == 1:
+            num_baros += 1
+        elif d.multi_id == 1:
             sensor_baro_1 = d.data
             print('found baro 1 data')
-            num_baros = 2
-        sensor_instance = sensor_instance +1
+            num_baros += 1
```
Comment (dagar) [inline Tools/process_sensor_caldata.py:95]: You might as well add sensor_baro 2. We're already carrying the parameters. https://github.com/PX4/Firmware/blob/c96b5246ffc50eeceef5aad0a31068a00f2b672b/src/modules/temperature_compensation/temp_comp_params_baro.c#L233-L319

**Comment by bresch** [comment]
>I'm not sure I follow, what do you mean "instance 1 appears before the 0"? Are you saying the instance numbering of the logged topic (ulg) is inconsistent with the system?

When looping through the topics:
```
for d in data:
    if d.name == 'sensor_accel':
```
The instance 1 of `sensor_accel` appeared before the instance 0

**Comment by dagar** [comment]
> > I'm not sure I follow, what do you mean "instance 1 appears before the 0"? Are you saying the instance numbering of the logged topic (ulg) is inconsistent with the system?
> 
> When looping through the topics:
> 
> ```
> for d in data:
>     if d.name == 'sensor_accel':
> ```
> 
> The instance 1 of `sensor_accel` appeared before the instance 0

Right ok, so even though we're not careful to advertise the drivers consistently we still have a race condition for first publication. 

Removing all ordering assumptions and always referencing device id is still the way to go, but we may want to look at log changes later to remove the potential discrepancy.

File: Tools/process_sensor_caldata.py (hunk 1 above)
Comment (bresch) [reply Tools/process_sensor_caldata.py:95]: I added that in https://github.com/PX4/Firmware/pull/14897/commits/f938b17b58defbc21f60cdcbfaa8348aaf0e175a . I followed the copy-paste pattern...

**Pull request merged**